src/Components/Homepage/HomepageList.js

```diff
@@ -0,0 +1,54 @@
+import React, { useEffect, useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+import { useSelector, useDispatch } from 'react-redux';
+import { getCompanies } from '../../redux/CompaniesList';
+import '../../styles/HomepageList.css';
+import HomepageItem from './HomepageItem';
+
+const HomepageList = () => {
+  const navigate = useNavigate();
+  const [search, setSearch] = useState('');
+  const { companyList } = useSelector((state) => state.companies);
+  const dispatch = useDispatch();
+  useEffect(() => {
+    dispatch(getCompanies());
+  }, [dispatch]);
+  return (
+    <div className="context">
+      <div className="search">
+        <input
+          className="input"
+          type="text"
+          onChange={(e) => setSearch(e.target.value)}
+          placeholder="Search Company"
+          value={search}
+        />
+      </div>
+
+      <div className="container">
+        {companyList
+          .filter((company) => {
+            if (search === '') {
+              return company;
+            } if (
+              company.symbol.toLowerCase().includes(search.toLowerCase())
+            ) {
+              return company;
+            }
+            return null;
+          })
+          .map((company) => (
+            <div
+              key={company.symbol}
+              onClick={() => navigate(`/details/${company.symbol}`)}
+              aria-hidden="true"
+            >
+              <HomepageItem title={company.symbol} />
+            </div>
+          ))}
+      </div>
+    </div>
+  );
+};
+
+export default HomepageList;
```

Comment:
- [ ] As required for the project, kindly make sure you follow the [design guidelines](https://www.behance.net/gallery/31579789/Ballhead-App-(Free-PSDs))  for your Homepage, as it should be close the to one from the template, including:
   * Typography: font face, size, and weight
   * Layout: `composition`, space between and around elements, text-alignment, color,...
   * Missing elements: `ShowRight Arrow` for each item card,...

| Template design | Your Design |
| ---------------------- | ----------------- | 
| ![Screenshot from 2023-01-12 23-41-09](https://github.com/Ashnaali3255/Stock-Metrics-WebApp/assets/83042742/fc8597d3-e795-410d-8881-12c45e3fd434) | ![Screenshot from 2023-06-06 21-48-00](https://github.com/Ashnaali3255/Stock-Metrics-WebApp/assets/83042742/b43cc118-d1bb-489b-85f4-b9aba3c943ff) |